languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,67 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="QUELLE_QUE" name="quelle que" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)?</token>
+                      <token>elle</token>
+                  </marker>
+                  <token regexp="yes">que|qu(&apostrophe;)</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelle</suggestion> ?</message>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> que soit sa situation.</example>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> qu'en soit la raison.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)?</token>
+                      <token>elles</token>
+                  </marker>
+                  <token regexp="yes">que|qu(&apostrophe;)</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelles</suggestion> ?</message>
+              <example correction="quelles">Il le fera <marker>qu'elles</marker> qu'en soient les conséquences.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token>quelle</token>
+                  <marker>
+                      <token>quelle</token>
+                  </marker>
+                  <token regexp="yes">soit</token>
+              </pattern>
+              <message>Une erreur de pronom semble être détectée. Souhaitez-vous écrire <suggestion>qu'elle</suggestion> ?</message>
+              <example correction="qu'elle">Donne-moi ta carte, quelle <marker>quelle</marker> soit.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)</token>
+                      <token>elle</token>
+                      <token regexp="yes">qu(&apostrophe;)</token>
+                  </marker>
+                  <token>un</token>
+              </pattern>
```

Comment:
There is an error in the rule. Remember the `<suggestion>` replaces all the `<token>`s in the `<marker>` section.

Take your first example:

input: Est-ce que //qu'elle qu'//un peut m'aider.

I've written the `<marker>` section //between slashes//. You replace that with the suggestion "quelqu'un" →

output: Est-ce que quelqu'unun peut m'aider.